rust/datafusion/src/physical_plan/hash_aggregate.rs

```diff
@@ -315,6 +318,41 @@ fn group_aggregate_batch(
     Ok(accumulators)
 }
 
+async fn compute_grouped_hash_aggregate(
```

Comment:
The main change in this file is to pull the computation of the output record batch into a future (aka an async function). I think this formulation is still a natural way to express the logic (you can see each input batch being fed to `group_aggregate_batch`)

Reply:
I also felt the need for this at some of my tries to simplify the hashAgg. code. 👍